src/wallet/wallet.cpp

```diff
@@ -1651,13 +1652,17 @@ CBlockIndex* CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, CBlock
     CBlockIndex* pindex = pindexStart;
     CBlockIndex* ret = nullptr;
     {
-        LOCK2(cs_main, cs_wallet);
         fAbortRescan = false;
         fScanningWallet = true;
 
         ShowProgress(_("Rescanning..."), 0); // show rescan progress in GUI as dialog or on splashscreen, if -rescan on startup
+        CBlockIndex* tip = nullptr;
+        {
+            LOCK(cs_main);
+            tip = chainActive.Tip();
+        }
         double dProgressStart = GuessVerificationProgress(chainParams.TxData(), pindex);
-        double dProgressTip = GuessVerificationProgress(chainParams.TxData(), chainActive.Tip());
+        double dProgressTip = GuessVerificationProgress(chainParams.TxData(), tip);
```

Comment:
GuessVerificationProcess needs cs_main (probably best to lock cs_main in GuessVerificationProcess).